mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -1416,6 +1440,177 @@ public struct Linger {
 		public int l_linger;
 	}
 
+	[Map]
+	[StructLayout (LayoutKind.Sequential)]
+	[CLSCompliant (false)]
+	public struct InAddr : IEquatable<InAddr> {
+		public uint s_addr;
+
+		public InAddr (byte b0, byte b1, byte b2, byte b3)
+		{
+			s_addr = 0;
+			this[0] = b0;
```

Comment:
This seems inefficient, as you'll be evaluating the `ArgumentOufOfRangeException` check *4* times, when you *know* you shouldn't need any.

I also wonder -- and I'm not sure how to verify :-( -- what byte ordering constraints there are, if any. I believe that you intend for `s_addr` to be in Network Byte Order...but should it be? I don't know what it should be, but this does give me pause.